dev/dev-env-setup.sh

```diff
@@ -0,0 +1,254 @@
+
+  The default environment setup is for basic functionality, installing the minimum development requirements dependencies.
+  To install the full development environment that supports working on all flavors and running all tests locally, set
+  the flag '-f' or '--full'
+
+-h, -help,        --help        Display help
+
+-d, -directory    --directory   The path to install the virtual environment into
+
+-f, -full         --full        Whether to install all dev requirements (Default: false)
+
+-v, -verbose      --verbose     Whether to fill stdout with every command or not
+
+-q, -quiet        --quiet       Whether to have pip install in quiet mode (Default: false)
+
+EOF
+}
+
+export verbose=0
+export quiet=0
+while getopts "d:fvqh" opt
+do
+  case "$opt" in
+    d) directory="$OPTARG" ;;
+    f) full_install=1 ;;
+    v) verbose=1 ;;
+    q) quiet=1 ;;
+    h) showHelp; exit ;;
+    *) showHelp; exit ;;
+  esac
+done
+
+if [[ $verbose == 1 ]]; then
+  set -exv
+fi
+
+# Acquire the OS for this environment
+case "$(uname -s)" in
+  Darwin*)                       machine=mac;;
+  Linux*)                        machine=linux;;
+  CYGWIN*|MINGW32*|MSYS*|MINGW*) machine=win;;
+  *)                             machine=unknown;;
+esac
+
+function quietpip(){
+  echo $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' )
+}
+
+# Check if pyenv is installed and offer to install it if not present
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  if [ -z "$MLFLOW_DEV_ENV_PYENV_INSTALL" ]; then
+    read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || $MLFLOW_DEV_ENV_PYENV_INSTALL == 1 ]]; then
+    if [[ "$machine" == mac ]]; then
+      # Check if brew is installed and install it if it isn't present
+      # Note: if xcode isn't installed, this will fail.
+      if [ -z "$(command -v brew)" ]; then
+        echo "Brew is required to install pyenv on MacOS. Installing in your home directory."
+        wget -O ~/brew_install.sh https://raw.githubusercontent.com/Homebrew/install/master/install.sh
+        bash ~/brew_install.sh
+      fi
+      echo "Updating brew and installing pyenv..."
+      echo "Note: this will probably take a considerable amount of time."
+      brew update
+      brew install pyenv
+    elif [[ "$machine" == linux ]]; then
+      sudo apt-get update -y
+      sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+      libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+      # Install pyenv from source
+      git clone --depth 1 https://github.com/pyenv/pyenv.git "$HOME/.pyenv"
+      PYENV_ROOT="$HOME/.pyenv"
+      PYENV_BIN="$PYENV_ROOT/bin"
+      if [ "$MLFLOW_DEV_ENV_CI_RUN" == 1 ]; then
```

Comment:
```suggestion
      if [ ! -z "$GITHUB_ACTIONS" ]; then
```

Can we use the `GITHUB_ACTIONS` environment variable instead?